examples/chef/common/clusters/switch/SwitchEventHandler.cpp

```diff
@@ -0,0 +1,88 @@
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/switch-server/switch-server.h>
+#include <app/server/Server.h>
+#include <app/util/att-storage.h>
+#include <platform/PlatformManager.h>
+
+#include "SwitchEventHandler.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Switch;
+using namespace chip::DeviceLayer;
+
+void SwitchEventHandler::OnSwitchLatched(EndpointId endpointId, uint8_t newPosition)
+{
+    ChipLogDetail(NotSpecified, "The latching switch is moved to a new position:%d", newPosition);
+
+    Clusters::SwitchServer::Instance().OnSwitchLatch(endpointId, newPosition);
+}
+
+void SwitchEventHandler::OnInitialPress(EndpointId endpointId, uint8_t newPosition)
+{
+    ChipLogDetail(NotSpecified, "The new position when the momentary switch starts to be pressed:%d", newPosition);
+
+    Clusters::SwitchServer::Instance().OnInitialPress(endpointId, newPosition);
+}
+
+void SwitchEventHandler::OnLongPress(EndpointId endpointId, uint8_t newPosition)
+{
+    ChipLogDetail(NotSpecified, "The new position when the momentary switch has been pressed for a long time:%d", newPosition);
+
+    Clusters::SwitchServer::Instance().OnLongPress(endpointId, newPosition);
+}
+
+void SwitchEventHandler::OnShortRelease(EndpointId endpointId, uint8_t previousPosition)
+{
+    ChipLogDetail(NotSpecified, "The the previous value of the CurrentPosition when the momentary switch has been released:%d",
+                  previousPosition);
+
+    Clusters::SwitchServer::Instance().OnShortRelease(endpointId, previousPosition);
+}
+
+void SwitchEventHandler::OnLongRelease(EndpointId endpointId, uint8_t previousPosition)
+{
+    ChipLogDetail(NotSpecified,
+                  "The the previous value of the CurrentPosition when the momentary switch has been released after having been "
+                  "pressed for a long time:%d",
+                  previousPosition);
+
+    Clusters::SwitchServer::Instance().OnLongRelease(endpointId, previousPosition);
+}
+
+void SwitchEventHandler::OnMultiPressOngoing(EndpointId endpointId, uint8_t newPosition, uint8_t count)
+{
+    ChipLogDetail(NotSpecified, "The new position when the momentary switch has been pressed in a multi-press sequence:%d",
+                  newPosition);
+    ChipLogDetail(NotSpecified, "%d times the momentary switch has been pressed", count);
+
+    Clusters::SwitchServer::Instance().OnMultiPressOngoing(endpointId, newPosition, count);
+}
+
+void SwitchEventHandler::OnMultiPressComplete(EndpointId endpointId, uint8_t previousPosition, uint8_t count)
+{
+    ChipLogDetail(NotSpecified, "The previous position when the momentary switch has been pressed in a multi-press sequence:%d",
+                  previousPosition);
+    ChipLogDetail(NotSpecified, "%d times the momentary switch has been pressed", count);
```

Comment:
Please make logging more compact for all logs in this file, like:

`"OnMultipressComplete(endpoint: %d, previousPosL %d, count: %d)"` or something like that, everywhere 

This will be easier to read than long English sentences.

Reply:
Fixed.